### PR TITLE
initial Lambda destination event

### PR DESCRIPTION
**Pull request by msailes**
*Issue #, if available:*
https://github.com/aws/aws-lambda-java-libs/issues/104

*Description of changes:*
Event to represent the Lambda destination event.

Maybe this should be called something else. I'm open to suggestions.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        private String requestId;
+        private String functionArn;
+        private String condition;
+        private int approximateInvokeCount;
+    }
+
+    @NoArgsConstructor
+    @Data
+    public static class ResponseContext implements Serializable, Cloneable {
+        private String statusCode;
```
Comment (andreac30) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java:52]: Looking at the AWS doc for the LambdaDestinationEvent I think statusCode should be an int type.

Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java]: Fixed.

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.amazonaws.services.lambda.runtime.events;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
```
Comment (andreac30) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java:16]: As the other events, I think also here the `NoArgsConstructor` is superfluous.

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        private int statusCode;
+        private String executedVersion;
+        private String functionError;
+    }
+
+    @NoArgsConstructor
+    @Data
+    public static class ResponsePayload implements Serializable, Cloneable {
+        private String errorMessage;
+    }
```
Comment (andreac30) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java:61]: Looking at [aws doc for lambda destination event](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-destinations/) it seems that `ResponsePayload` and `ResponseContext` can have different fields. I would suggest using Map<String, Object> as the `requestPayload` above, what do you think?

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java (hunk 2 above)
Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java]: fixed

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java (hunk 3 above)
Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/LambdaDestinationEvent.java]: fixed

**Comment by msailes** [comment]
I've just tested this on my account.

**Pull request merged**